httpcore5/src/main/java/org/apache/hc/core5/http/impl/io/BHttpConnectionBase.java

```diff
@@ -147,13 +149,20 @@ protected OutputStream createContentOutputStream(
         if (len >= 0) {
             return new ContentLengthOutputStream(buffer, outputStream, len);
         } else if (len == ContentLengthStrategy.CHUNKED) {
-            final int chunkSizeHint = http1Config.getChunkSizeHint() >= 0 ? http1Config.getChunkSizeHint() : 2048;
-            return new ChunkedOutputStream(buffer, outputStream, chunkSizeHint, trailers);
+            return new ChunkedOutputStream(buffer, outputStream, getChunkedRequestBuffer(), trailers);
         } else {
             return new IdentityOutputStream(buffer, outputStream);
         }
     }
 
+    private byte[] getChunkedRequestBuffer() {
+        if (chunkedRequestBuffer == null) {
+            final int chunkSizeHint = this.http1Config.getChunkSizeHint();
+            chunkedRequestBuffer = new byte[chunkSizeHint > 0 ? chunkSizeHint : 8192];
```

Comment:
I think this should still be `>= 0` to avoid changing behavior when zero is set, allowing each write to produce a chunk.

Reply:
On second review this matches the logic in ChunkedOutputStream which used the default buffer size when chunkSizeHint was set to zero.